controller/common/lut.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+from scipy import interpolate
+
+
+class LUT:
+    """
+    Class for performing look-up table interpolation.
+
+    Methods:
+        __init__: Initializes the LUT object with lookup table data and interpolation method.
+        __call__: Calls the interpolation method with the given input.
+    """
+
+    def __init__(self, lookup_table: list, interpolation_method: str = "linear"):
+        """
+        Initializes the LUT object.
+
+        Args:
+            lookup_table (list): A list of tuples or lists containing x-y data points for
+            interpolation. Format is [[x-values],[y-values]]
+            interpolation_method (str): Interpolation method to use. Default is "linear".
+
+        Raises:
+            ValueError: If the specified interpolation method is unknown.
+        """
+        self.__table = np.array(lookup_table)
```

Comment:
You should probably check the validity of the input. For example, is it 2D? Is there an x to y mapping for each input? For example, I could break your implementation with:

`[1,2,3]`,  or `[[1,2], [3]]`

Reply:
added verify function:
```
def verifyTable(self, table: np.ndarray) -> None:
        if table.shape[1] == 2 and len(table.shape) == 2:
            return
        else:
            raise ValueError("Input table is invalid shape.")
```

added verify function to `___init___`:

```
self.__table = np.array(lookup_table)
self.verifyTable(self.__table)
        
```